TAR archive creation fails because of GUID problems

## Situation
A member in our team mentioned following problem:

```
java -jar sechub-pds-tools-cli-1.0.0.jar generate -p test/sechub.json -s codeScan
```

sechub.json:
```json
{
  "apiVersion": "1.0",
  "codeScan": {
    "use": [
    "test-sources"
    ]
  },
  "data": {
    "binaries": [
     {
       "name": "test-sources",
       "fileSystem" : {
          "folders" : [ "some-docker-container.tar" ]
       }
     }
    ]
  }
}
```

#### Output

Error:
```java
11:13:45.840 [main] DEBUG com.mercedesbenz.sechub.commons.TextFileWriter - Written: /tmp/pds_solution_gen3139246774933591280/original-used-sechub-configfile.json
11:13:45.848 [main] DEBUG com.mercedesbenz.sechub.commons.TextFileWriter - Written: /tmp/pds_solution_gen3139246774933591280/reducedSecHubJson_for_codeScan.json
11:13:45.850 [main] DEBUG com.mercedesbenz.sechub.commons.TextFileWriter - Written: /tmp/pds_solution_gen3139246774933591280/pdsJobData.json
11:13:45.852 [main] DEBUG com.mercedesbenz.sechub.commons.archive.ArchiveSupport - Start creating TAR archive: /tmp/pds_solution_gen3139246774933591280/binaries.tar
Generation failed: Creation of archives failed
Exception in thread "main" java.io.IOException: Creation of archives failed
at com.mercedesbenz.sechub.commons.archive.ArchiveSupport.createArchives(ArchiveSupport.java:139)
at com.mercedesbenz.sechub.pds.tools.generator.PDSSolutionTestFilesGenerator.generate(PDSSolutionTestFilesGenerator.java:94)
at com.mercedesbenz.sechub.pds.tools.generator.PDSSolutionTestFilesGenerator.generate(PDSSolutionTestFilesGenerator.java:67)
at com.mercedesbenz.sechub.pds.tools.PDSToolsCLI.generate(PDSToolsCLI.java:104)
at com.mercedesbenz.sechub.pds.tools.PDSToolsCLI.start(PDSToolsCLI.java:68)
at com.mercedesbenz.sechub.pds.tools.PDSToolsCLI.main(PDSToolsCLI.java:33)
Caused by: java.lang.IllegalArgumentException: group id '89636213' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit
at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.failForBigNumber(TarArchiveOutputStream.java:406)
at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.failForBigNumberWithPosixMessage(TarArchiveOutputStream.java:428)
at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.failForBigNumbers(TarArchiveOutputStream.java:414)
at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.putArchiveEntry(TarArchiveOutputStream.java:592)
at com.mercedesbenz.sechub.commons.archive.ArchiveSupport.compressRecursively(ArchiveSupport.java:398)
at com.mercedesbenz.sechub.commons.archive.ArchiveSupport.createArchive(ArchiveSupport.java:211)
at com.mercedesbenz.sechub.commons.archive.ArchiveSupport.createArchives(ArchiveSupport.java:122)
```

## Wanted
That shall not happen

## Analyse
- https://commons.apache.org/proper/commons-compress/tar.html
  (see Big Numeric Values)

## Solution
We will use `TarArchiveOutputStream.BIGNUMBER_POSIX` for big numbers

--- a/sechub-commons-archive/src/main/java/com/mercedesbenz/sechub/commons/archive/ArchiveSupport.java
+++ b/sechub-commons-archive/src/main/java/com/mercedesbenz/sechub/commons/archive/ArchiveSupport.java
@@ -168,6 +168,7 @@ public class ArchiveSupport {
                 @SuppressWarnings("resource")
                 TarArchiveOutputStream tarOutputStream = (TarArchiveOutputStream) outputStream;
                 tarOutputStream.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
+                tarOutputStream.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_POSIX);
             }
 
             Map<String, CreationPathContext> uniqueNameToPaths = creationContext.getUniqueToPathsMap();

